apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -535,3 +536,41 @@ export async function validateContactManager(contactManagerUserId: string) {
 		throw new Error('error-contact-manager-not-found');
 	}
 }
+
+export async function unverifyContactChannel(
+	contactId: ILivechatContact['_id'] | undefined,
+	channelName: string,
+	visitorId: string,
+): Promise<ILivechatContact> {
+	if (!contactId) {
+		throw new Error('error-invalid-contact');
+	}
+
+	const contact: Pick<ILivechatContact, '_id' | 'channels'> | null = await LivechatContacts.findOneById(contactId, {
+		projection: {
+			_id: 1,
+			channels: 1,
+		},
+	});
+
+	if (!contact) {
+		throw new Error('error-invalid-contact');
+	}
+
+	if (!contact.channels) {
+		throw new Error('error-invalid-channel');
+	}
+
+	const channelToUnverify = contact.channels.findIndex((channel) => channel.name === channelName && channel.visitorId === visitorId);
+	if (channelToUnverify < 0) {
+		throw new Error('error-invalid-channel');
+	}
+
+	contact.channels[channelToUnverify].verified = false;
+
+	return LivechatContacts.updateContact(contact._id, {
+		channels: contact.channels,
+	});
```

Comment:
You can do all of this with a single mongo operation:
```
updateOne({ _id: contact._id, 'channels.visitorId': visitorId, 'channels.channelName': channelName }, { $set: { 'channels.$.verified': false } });
```

Then just check the result to determine if any channel was updated or not.